manim/utils/color/core.py

```diff
@@ -1508,9 +1508,95 @@ def random_color() -> ManimColor:
     ManimColor
         A random :class:`ManimColor`.
     """
-    import manim.utils.color.manim_colors as manim_colors
+    return RandomColorGenerator().next()
 
-    return random.choice(manim_colors._all_manim_colors)
+
+class RandomColorGenerator:
+    """
+    A generator for producing random colors from a given list of Manim colors,
+    optionally in a reproducible sequence using a seed value.
+
+    When initialized with a specific seed, this class produces a deterministic
+    sequence of `ManimColor` instances. If no seed is provided, the selection is
+    non-deterministic using Python’s global random state.
+
+    Parameters
+    ----------
+    seed : int | None, optional
+        A seed value to initialize the internal random number generator.
+        If None (default), colors are chosen using the global random state.
+
+    sample_colors : list[ManimColor], optional
+        A custom list of Manim colors to sample from. Defaults to the full Manim color palette.
+
+    Examples
+    --------
+    Without a seed (non-deterministic):
```

Comment:
The actual formatting for the example blocks need to be slightly adapted to make the documentation render correctly. I'll push a commit to take care of it. 👍